### PR TITLE
Convert SquooshTreeBuilder.kt

### PR DESCRIPTION
**Pull request by yiqunw700**
Fixes: #1912

### CODE REVIEW

File: designcompose/src/main/java/com/android/designcompose/squoosh/SquooshTreeBuilder.kt [hunk 1]
```diff
@@ -192,22 +182,24 @@ internal fun resolveVariantsRecursively(
...
         // First ask the interaction state if there's a variant we should render instead.
-        view = interactionState.squooshNodeVariant(v.id, customizations.getKey(), document) ?: v
+        interactionState.squooshNodeVariant(v.id, customizations.getKey(), document)?.let {
+            view = it
+        }
 
         // If we didn't replace the component because of an interaction, we might want to replace it
         // because of a variant customization.
-        if (view.name == v.name) {
+        if (view.name == v.name) { // TODO: why we don't check if view == v????
```
Comment (yiqunw700) [inline designcompose/src/main/java/com/android/designcompose/squoosh/SquooshTreeBuilder.kt:202]: Not sure if this means `view == v` and below if I can assume `view.componentInfo` is actually the same with `v.componentInfo`

File: designcompose/src/main/java/com/android/designcompose/squoosh/SquooshTreeBuilder.kt [hunk 2]
```diff
@@ -488,58 +479,30 @@ internal fun generateReplacementListChildNode(
...
-                paged_scrolling = false
-                overflow = OverflowDirection.None().toInt()
-            }
-            .build()
-
-    val listChildView = View.Builder()
-    listChildView.unique_id = 0 // This is unused.
-    listChildView.id = "replacement-${node.view.id}-${childIdx}"
-    listChildView.name = "Replacement List ${node.view.name} / $childIdx"
-    listChildView.component_info = Optional.empty()
```
Comment (yiqunw700) [inline designcompose/src/main/java/com/android/designcompose/squoosh/SquooshTreeBuilder.kt:522]: Not sure if it works without setting any value to the `componentInfo` which used to be Optional.empty()

File: designcompose/src/main/java/com/android/designcompose/squoosh/SquooshTreeBuilder.kt [hunk 3]
```diff
@@ -192,22 +181,24 @@ internal fun resolveVariantsRecursively(
...
-        if (componentInfo.overrides.isPresent) {
-            val overrides = componentInfo.overrides.get()
-            if (overrides.style.isPresent) {
-                overrideStyle = overrides.style.get()
+        if (v.componentInfo.hasOverrides()) {
+            val overrides = v.componentInfo.overrides
+            if (overrides.hasStyle()) {
+                overrideStyle = overrides.style
             }
         }
```
Comment (timothyfroehlich) [inline designcompose/src/main/java/com/android/designcompose/squoosh/SquooshTreeBuilder.kt:189]: Can simplify this to 
```suggestion
        overrideStyle = v.componentInfoOrNull?.overridesOrNull?.styleOrNull
```

File: designcompose/src/main/java/com/android/designcompose/squoosh/SquooshTreeBuilder.kt (hunk 1 above)
Comment (timothyfroehlich) [reply designcompose/src/main/java/com/android/designcompose/squoosh/SquooshTreeBuilder.kt:201]: This code really isn't clear, it's hard to tell what `v` and `view` are. @rylin8, could you take a look at this and maybe clean it up with some better variable nmames?

File: designcompose/src/main/java/com/android/designcompose/squoosh/SquooshTreeBuilder.kt (hunk 2 above)
Comment (timothyfroehlich) [reply designcompose/src/main/java/com/android/designcompose/squoosh/SquooshTreeBuilder.kt:522]: It works, new messages have all fields unset.

File: designcompose/src/main/java/com/android/designcompose/squoosh/SquooshTreeBuilder.kt [hunk 4]
```diff
@@ -488,58 +479,30 @@ internal fun generateReplacementListChildNode(
...
-        Optional.of(ViewData(Optional.of(ViewDataType.Container(listChildViewData))))
-    listChildView.design_absolute_bounding_box = Optional.empty()
-    listChildView.render_method = RenderMethod.None().toInt()
-    listChildView.explicit_variable_modes = emptyMap()
+    val listChildView = view {
+        uniqueId = 0 // This is unused.
+        id = "replacement-${node.view.id}-${childIdx}"
+        name = "Replacement List ${node.view.name} / $childIdx"
+        scrollInfo = listChildScrollInfo
+        data = viewData { this.container = listChildViewData }
```
Comment (timothyfroehlich) [inline designcompose/src/main/java/com/android/designcompose/squoosh/SquooshTreeBuilder.kt:494]: Unnecessary this
```suggestion
        data = viewData { container = listChildViewData }
```

**Review by timothyfroehlich — CHANGES_REQUESTED** [review]

File: designcompose/src/main/java/com/android/designcompose/squoosh/SquooshTreeBuilder.kt (hunk 1 above)
Comment (rylin8) [reply designcompose/src/main/java/com/android/designcompose/squoosh/SquooshTreeBuilder.kt]: Done

File: designcompose/src/main/java/com/android/designcompose/squoosh/SquooshTreeBuilder.kt (hunk 3 above)
Comment (rylin8) [reply designcompose/src/main/java/com/android/designcompose/squoosh/SquooshTreeBuilder.kt]: Done

File: designcompose/src/main/java/com/android/designcompose/squoosh/SquooshTreeBuilder.kt (hunk 4 above)
Comment (rylin8) [reply designcompose/src/main/java/com/android/designcompose/squoosh/SquooshTreeBuilder.kt]: Done

File: designcompose/src/main/java/com/android/designcompose/squoosh/SquooshTreeBuilder.kt [hunk 5]
```diff
@@ -220,14 +211,14 @@ internal fun resolveVariantsRecursively(
             val variantNodeName =
                 variantTransition.selectVariant(
                     v.id,
-                    customizations.getMatchingVariant(view.component_info) ?: componentInfo.name,
+                    customizations.getMatchingVariant(v.componentInfo) ?: v.componentInfo.name,
```
[the remark this answers is not in the document]
Comment (rylin8) [reply designcompose/src/main/java/com/android/designcompose/squoosh/SquooshTreeBuilder.kt]: Done

File: designcompose/src/main/java/com/android/designcompose/squoosh/SquooshTreeBuilder.kt [hunk 6]
```diff
@@ -220,14 +211,14 @@ internal fun resolveVariantsRecursively(
...
                 )
-            if (variantNodeName != null && variantNodeName != componentInfo.name) {
+            if (variantNodeName != null && variantNodeName != v.componentInfo.name) {
                 // Find the view associated with the variant name
                 val variantNodeQuery =
                     NodeQuery.NodeVariant(
                         variantNodeName,
-                        variantParentName.ifEmpty { view.component_info.get().component_set_name },
+                        variantParentName.ifEmpty { v.componentInfo.componentSetName },
```
[the remark this answers is not in the document]
Comment (rylin8) [reply designcompose/src/main/java/com/android/designcompose/squoosh/SquooshTreeBuilder.kt]: Done

**Review by timothyfroehlich — APPROVED** [review]

**Pull request merged**